### ISSUE TITLE
[sources/parsley] Update parsley source to generate fake GPS messages for testing

### ISSUE THREAD

**Issue by EmpiricalCode** [waterloo-rocketry/omnibus#315]
# Feature Description
- **Expected implementation date:** 2024-11-9
- **As a:** Developer for omnibus, 
- **I would want to:** generate fake GPS CAN messages from the parsley source.
- **Because:** fake GPS CAN messages are needed for testing new GPS sinks.

## Related Links
[ #265](https://github.com/waterloo-rocketry/omnibus/pull/265)

## Details
Note that the original fake messages have been removed as they are outdated. In a future feature request new fake messages should be added to the parsley source. 

Note that this feature will only work once a small change to the parsley repo has been made to support the new CAN 2.0 format (I will make a PR for this)

---

# Requirements
## Specific Requirements
- Generated GPS messages MUST comply with CAN Bus 2.0 format.
- For longitude/latitude, generated coordinates MUST lie within the general Waterloo region.
- The timestamp MUST wrap around once it exceeds the maximum 3 bytes.

## Standard Requirements - Do Not Modify Without Approval
- All changes made MUST be tested and reviewed by a code reviewer.
- The conditions outlined in CONTRIBUTING.MD MUST be respected.
- Deep consider of new library dependencies MUST be made before adding them to the project.
- Keep the codebase clean, don't push non-code files (e.g. DATA, VirtualEnv, etc.) to the repository.
- All new feature changes made SHOULD have associated unit tests (if applicable)
- Keep well-documented code, and follow the style guide.

Satisfying the above requirements WILL close this issue.

# Stakeholder / Product Owner
Software Team

**Comment by ChrisYx511** [comment]
@EmpiricalCode I think the fake GPS data is actually going to be used by software dev people to test GPS functionality and NOT for DAQ people, as DAQ doesn't need fake GPS coordinates since it will get them from real parsley, @zangjiucheng can you confirm?

**Comment by ChrisYx511** [comment]
but also yay fake GPS data!! 😎 🎉🎉 🎉 🎉

**Comment by zangjiucheng** [comment]
Yep, you are right :)) This is for software team easier for UI development but don't need to download large dataset. DAQ system they don't need to care with this.

**Comment by zangjiucheng** [comment]
It not a big deal, not mandatory require to change the issue description, but just clarify.

**Comment by EmpiricalCode** [comment]
Got it, will leave as "Software Team"

**Comment by ChrisYx511** [comment]
Descoped.
